java-frontend/src/main/java/org/sonar/java/cfg/VariableReadExtractor.java

```diff
@@ -44,19 +49,31 @@ public LocalVariableReadExtractor(Symbol.MethodSymbol methodSymbol) {
 
   @Override
   public void visitAssignmentExpression(AssignmentExpressionTree tree) {
-    //skip writing to a variable.
-    if(!tree.variable().is(Tree.Kind.IDENTIFIER)) {
+    //skip writing to a variable or field.
+    if(!tree.variable().is(Tree.Kind.IDENTIFIER) && !tree.variable().is(Kind.MEMBER_SELECT)) {
       scan(tree.variable());
     }
     scan(tree.expression());
   }
 
   @Override
   public void visitIdentifier(IdentifierTree tree) {
-    if(methodSymbol.equals(tree.symbol().owner())) {
+    Symbol owner = tree.symbol().owner();
+    if(methodSymbol.equals(owner) || (includeFields && methodSymbol.owner().equals(owner))) {
       used.add(tree.symbol());
     }
     super.visitIdentifier(tree);
   }
 
+  @Override
+  public void visitMemberSelectExpression(MemberSelectExpressionTree tree) {
+    Symbol owner = tree.identifier().symbol().owner();
```

Comment:
I don't see how this is covered by test (you don't have memberSelect access to a field in your tests). Moreover I think that given what you've implemented in visitIdentifier this whole method should be removed as the identifier will be visited anyway : right now this would result in adding the field symbol twice in the `used` list.